File: android/wear/src/main/java/app/until/time/watch/PeriodSnapshot.kt
package app.until.time.watch

data class PeriodSnapshot(
    val progress: Double,
    val percentDone: Int,
    val percentLeft: Int,
    val startMs: Long?,
    val endMs: Long?,
    val remainingLabel: String,
    val passedLabel: String,
    val whisper: String,
    val updatedAt: Long,
) {
    val progressClamped: Float
        get() = progress.toFloat().coerceIn(0f, 1f)
}

File: ios/UNTILWatchShared/PeriodSnapshot.swift
import Foundation

struct PeriodSnapshot: Equatable {
  var progress: Double
  var percentDone: Int
  var percentLeft: Int
  var remainingLabel: String
  var passedLabel: String
  var updatedAt: TimeInterval

  var progressClamped: Double {
    min(1, max(0, progress))
  }
}
